bindings/nodejs/lib/index.ts

```diff
@@ -41,42 +52,95 @@ export * from './wallet';
 export * from './logger';
 
 // For future reference to see what we return from rust as a serialized string
-export type Result = {
-    // "error" | "panic", or other binding "Response" enum name, we consider "ok".
+export type Result = OkResult | NotOkResult;
+
+export type OkResult = {
+    // "Response" enum name, we consider "ok".
     type: string;
-    // "panic" means payload is just a string, otherwise its the object below.
+    // The object below.
     payload: {
         // Ok: All method names from types/bridge/__name__.name
-        // Not ok: all variants of iota_sdk_bindings_core::Error type i.e block/client/wallet/
         type: string;
-        // If "ok", json payload
-        payload?: string;
-        // If not "ok", json error
-        error?: string;
+        // json payload
+        payload: string;
     };
 };
 
+export type NotOkResult = PanicResult | ErrorResult;
+
+export type ErrorResult = {
+    type: 'error';
+    payload: {
+        // All variants of iota_sdk_bindings_core::Error type i.e block/client/wallet.
+        // client and wallet have a full object, the others a string.
+        type: string;
+        // json error
+        error:
+            | {
+                  // all variants of each sub-error type (i.e. healthyNodePoolEmpty )
+                  type: string;
+                  // Error message from the enum.to_string() generation
+                  error: string;
+              }
+            | string;
+    };
+};
+
+export type PanicResult = {
+    type: 'panic';
+    payload: string;
+};
+
 function errorHandle(error: any): Error {
     try {
-        const err: Result = JSON.parse(error.message);
-        if (!err.type) {
+        let err: Result = JSON.parse(error.message);
+        if (!err.type || !(err.type == 'panic' || err.type == 'error')) {
             return error;
         }
 
+        // Guaranteed error
+        err = err as NotOkResult;
```

Comment:
Not sure what the purpose of this is

Reply:
have to force the type, otherwise it warns that payload can be string or object